### PR TITLE
NETOBSERV-90: enable deployment as Daemonset

### PR DESCRIPTION
**Pull request by mariomac**
- Adding a HostPort to the container specification in the DaemonSet
- Setting the proper RBAC permissions
- Enable the deployment in the master node

### CODE REVIEW

File: controllers/flowcollector_controller.go [hunk 1]
```diff
@@ -172,7 +173,6 @@ func (r *FlowCollectorReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&corev1.ConfigMap{}).
 		Owns(&appsv1.Deployment{}).
 		Owns(&appsv1.DaemonSet{}).
-		Owns(&corev1.Service{}).
```
Comment (OlivierCazade) [inline controllers/flowcollector_controller.go:175]: Why removing this ?

**Review by OlivierCazade — APPROVED** [review]
LGTM, my only comment is more a question than a comment

File: controllers/flowcollector_controller.go (hunk 1 above)
Comment (mariomac) [reply controllers/flowcollector_controller.go:175]: It is duplicated 2 lines below.

Comment (OlivierCazade) [reply controllers/flowcollector_controller.go:175]: Thanks

File: controllers/goflowkube/goflowkube_objects.go [hunk 2]
```diff
@@ -72,18 +71,33 @@ func buildDaemonSet(desired *flowsv1alpha1.FlowCollectorGoflowKube, ns string) *
...
 	}
 }
 
-func buildPodTemplate(desired *flowsv1alpha1.FlowCollectorGoflowKube) *corev1.PodTemplateSpec {
+func buildPodTemplate(desired *flowsv1alpha1.FlowCollectorGoflowKube, forDaemonSet bool) corev1.PodTemplateSpec {
 	cmd := buildMainCommand(desired)
-	return &corev1.PodTemplateSpec{
+	var ports []corev1.ContainerPort
+	var tolerations []corev1.Toleration
+	if forDaemonSet {
```
Comment (jpinsonneau) [inline controllers/goflowkube/goflowkube_objects.go:83]: can't we use `desired.Kind == constants.DaemonSetKind` here instead of an additionnal parameter ?

**Review by jpinsonneau — CHANGES_REQUESTED** [review]

File: controllers/goflowkube/goflowkube_objects.go (hunk 2 above)
Comment (mariomac) [reply controllers/goflowkube/goflowkube_objects.go:83]: Sure. Changing

**Review by jpinsonneau — APPROVED** [review]
LGTM

**Pull request merged**